tests/integration/test_state_result.py

```diff
@@ -0,0 +1,98 @@
+from typing import no_type_check
+
+from guppylang import decorator
+from guppylang.module import GuppyModule
+from guppylang.std.builtins import state_result, array, owned
+from guppylang.std.quantum import qubit, discard, x, h
+from guppylang.std import quantum
+
+from tests.integration.test_quantum import compile_quantum_guppy
+
+
+def test_basic(validate):
+    @compile_quantum_guppy
+    def main() -> None:
+        q1 = qubit()
+        q2 = qubit()
+        state_result("tag", q1, q2)
+        discard(q1)
+        discard(q2)
+
+    validate(main)
+
+
+def test_empty(validate):
+    @compile_quantum_guppy
+    def main() -> None:
+        state_result("tag")
```

Comment:
Not sure if we should allow this. I could see users writing this with the assumption that the *whole state* will be outputted because they are not aware that you have to pass the qubits.

Since empty state results are useless anyways, maybe it would be more helpful to add a diagnostic along the lines of

```
2 |     state_result("tag")
  |                       ^ Qubits whose state should be reported must be passed explicitly
```